### DISCUSSION TITLE
Many task and one call at a time

### DISCUSSION THREAD

**Discussion by darlandieterich** [go-co-op/gocron#109] [Q&A] [answered]
I have two request in different times, but when have 1 minute two jobs are called and the server has reached its maximum number of sessions. I don't wanna upgrade the session in my request server, i want manage the call one call at a time, without conflict.

```go
s := gocron.NewScheduler(time.UTC)
_, _ = s.Every(30).Seconds().Do(getX) 
_, _ = s.Every(60).Seconds().Do(getY) 
s.StartBlocking()
```

**Comment by JohnRoesler** [comment]
Hey @darlandieterich - sounds like you'd like to be able to supply a custom `limiter` to the job? Am I understanding right? The idea being you'd be able to limit the jobs based on available sessions in this case?

**Comment by JohnRoesler** [comment, marked as answer]
This can be accomplished with https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.SetMaxConcurrentJobs

Use reschedule mode https://pkg.go.dev/github.com/go-co-op/gocron#pkg-constants

For example:

```go
package main

import (
	"fmt"
	"time"

	"github.com/go-co-op/gocron"
)

func main() {
	s := gocron.NewScheduler(time.UTC)
	s.SetMaxConcurrentJobs(1, gocron.RescheduleMode)
	_, _ = s.Every(1).Seconds().Do(func() {
		fmt.Println("This will run once every 5 seconds even though it is scheduled every second because maximum concurrent job limit is set.")
		time.Sleep(5 * time.Second)
	})
}

```
